ftplugin/markdown/quicklink.vim

```diff
@@ -95,4 +95,45 @@ function! OpenLinkOnCurrentLine()
   call system('open ' . expand('<cWORD>'))
 endfunction
 
+function! s:OnMarkdownLink()
+    return synIDattr(synID(line("."),col("."),1),"name") == "markdownLinkText" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "markdownLinkTextDelimiter" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "mkdLink" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "mkdDelimiter"
+endf
+
+function! s:OpenMarkdownLink()
+  "get the initial position of the curosr
+  let initial_pos = getpos('.')
+  "empty the @a register in the case cursor is not well palced and that old
+  "patterns would still trigger.
+  let @a = ""
+  normal "ayi]
+  let escaped_link_name = escape(getreg('a'), '&')
+  let link_target_pattern = '\v^\['.escaped_link_name.'\]: (%(ftp[s]?|http[s]?):\/\/\S+)>'
+  " go to link location
+  let found = search(link_target_pattern, 'e')
+  if found
+    "use netrw-gx to open the link
+    call netrw#BrowseX(expand("<cfile>"),0)
+  endif
+  call setpos('.', initial_pos)
+endfunction
+
+function! s:MarkdownAwareGX()
+  if s:OnMarkdownLink()
+    call s:OpenMarkdownLink()
+  else
+    call netrw#BrowseX(expand("<cfile>"),0)
+  endif
+endfunction
+
+command! MarkdownAwareGX call <sid>MarkdownAwareGX()
+if !exists('g:quicklink_open_mapping')
+  let g:quicklink_open_mapping = 'gx'
+endif
+if g:quicklink_open_mapping != ''
+  execute 'nnoremap <buffer> '.g:quicklink_open_mapping.' :MarkdownAwareGX<cr>'
+endif
```

Comment:
Now that we are only augmenting `gx` rather than providing a distinct mapping, I'd prefer to just use that. If user's want a custom map they can use the command you've defined on 131. Then these lines can just be:

``` vim
command! MarkdownAwareGX call <sid>MarkdownAwareGX()
nnoremap <buffer> gx :MarkdownAwareGX<cr>
```